src/README.md

```diff
@@ -43,7 +43,7 @@ in Openshift can be achieved with the following step:
 The same application can also be built using the standalone [S2I](https://github.com/openshift/source-to-image) application on systems that have it available:
 
     ```
-    $ s2i build https://github.com/sclorg/s2i-python-container.git --context-dir={{ spec.version }}/test/setup-test-app/ rhscl/python-{{ spec.short_ver }}-rhel7 python-sample-app
+    $ s2i build https://github.com/sclorg/s2i-python-container.git --context-dir={{ spec.version }}/test/setup-test-app/ <image_name> python-sample-app
```

Comment:
The reader doesn't know what image_name format to use. "rhscl/python-38-rhel7" is not a very intuitive name. Can we list somewhere the possible image names for example?

Reply:
The README.md is a jinja2 template. README.md alone is generated by distgen. But agree with @torsava <image_name> is not known. Examples would be fine.

Reply:
There is no specific list of names because it depends on when you download an image from. Or you can also build it on your own and then the name might be completely different.
What about something like this: `Where <image_name> is the s2i-python image you downloaded from RHEL, Centos or Fedora registry or built from these sources.` And the "downloaded" and "built" might be links to the main readme where links to registries are provided. What do you think about it?

Reply:
I myself lean a bit towards the older format (only listing rhel7 images). Sure, it might not be usable for every version (maybe add a note somewhere?) but at least the user knows what image name to use.
If you want to go with your proposal I would agree with Tomas - you need to list the image names somewhere so the user does not have to look up what to use. Which means you need to generate them with distgen. At that point, why not just generate a working example in the first place (and list alternatives)?

Reply:
First of all, there is no single example working with all currently supported Python versions. That's the problem I am trying to solve.
The problem with names and examples is that it might vary. For example, when I download `python-36-centos7` from docker hub, podman shows me this image with the full name `docker.io/centos/python-36-centos7`. I can use it by the full name, or just `centos/python-36-centos7` or the shortest `python-36-centos7`. But I cannot do the same with Fedora python3 image because the shortest name does not work.

I can update the proposed note this way (links do not work yet):
Where <image_name> is the s2i-python image you [downloaded from RHEL, Centos or Fedora registry]() or [built]() from these sources. For example `ubi8/python-36`, `centos/python-36-centos7` or `f31/python3`.

Reply:
>First of all, there is no single example working with all currently supported Python versions. That's the problem I am trying to solve.

There are working examples for some versions. For the others a substitute needs to happen first (for a Fedora or Centos image name). After this change, there are no working examples for any versions as a substitute needs to happen for all of them. To me, this seems more like a downgrade.
But again, I do not want to block you from doing it, just providing some feedback.

>But I cannot do the same with Fedora python3 image because the shortest name does not work.
This is interesting and I never noticed. If this is important maybe we should look into why this does not happen for Fedora images? Though in general I would opt for fully qualified names everywhere, as that is the only way to point to a correct image.

Reply:
I think that users of S2I images are experienced enough to do substitution and nobody should really use copy-pasted code examples.
From my point of view: as a user, I'd prefer a command that fails which reminds me to do a substitution instead of examples that work but might do something I don't want. Even accidentally when I copy a newline with the command and it is executed.

We can discuss it more tomorrow at the meeting.

Reply:
> I think that users of S2I images are experienced enough to do substitution and nobody should really use copy-pasted code examples.

Absolutely. But let's at least point them to the list of where they can download the images from that's on the min README?

Reply:
> Absolutely. But let's at least point them to the list of where they can download the images from that's on the min README?

Yes, in the latest proposal, there is a plan to add a link to the Download section in the main Readme and also a link to the Build section.

Reply:
I am going to commit that, so we can discuss it directly and add suggestions.

Reply:
Please, take a look at the new note below the example code.